File: tests/TypeOfNew.cs
using System;

namespace Demo {
    public partial class TypeOfNew {
        public static byte[] MakeArray(int len) {
            byte[] result;
            result = new byte[len];
            return result;
        }
        public static void Show() {
            Console.WriteLine(typeof(int));
        }
    }
}

File: tests/TypeOfNew.pas
namespace Demo;

type
  TypeOfNew = class
  public
    class method MakeArray(len: Integer): array of Byte;
    class method Show;
  end;

implementation
uses System;

class method TypeOfNew.MakeArray(len: Integer): array of Byte;
begin
  result := new Byte[len];
end;

class method TypeOfNew.Show;
begin
  Console.WriteLine(typeof(Integer));
end;

end.
